gsy_framework/sim_results/electric_blue/all_results.py

```diff
@@ -25,26 +26,34 @@ def __init__(self):
         self.asset_time_series: Dict[
             int, Dict[int, Dict[DateTime, Dict[str, Dict]]]] = defaultdict(
             lambda: defaultdict(lambda: defaultdict(dict)))
+        self.asset_volume_time_series: Dict[
+            str, Dict[AggregationResolution, AssetVolumeTimeSeries]] = {}
+        self.get_asset_volume_time_series_db = get_asset_volume_time_series_db
         self._total_memory_utilization_kb = 0.0
 
-    def update(self, _area_dict: Dict, core_stats: Dict, current_market_slot: str) -> None:
+    def update(self, area_dict: Dict, core_stats: Dict, current_market_slot: str) -> None:
         """
         Update the forward market aggregated results with bids_offers_trades of current slot.
         """
         if not current_market_slot:
             return
         self._buffer_current_asset_stats()
         self._clear_asset_stats()
-        for area_uuid, area_result in core_stats.items():
-            if "forward_market_stats" not in area_result:
-                return
-
-            current_market_dt = str_to_pendulum_datetime(current_market_slot)
-            for market_type_value, market_stats in area_result["forward_market_stats"].items():
-                market_type = int(market_type_value)
-                self._buffer_bids_offers_trades(area_uuid, market_type, market_stats)
-                current_results = handle_forward_results(current_market_dt, market_stats)
-                self._update_stats_and_time_series(current_results, market_type)
+
+        forward_results = None
+        for area_result in core_stats.values():
+            if "forward_market_stats" in area_result:
+                forward_results = area_result
+                break
```

Comment:
NIT: Could be refactored into a separate method

Reply:
Nice suggestion. Fixed it.